src/index.d.ts

```diff
@@ -46,7 +46,7 @@ export interface Color {
  */
 export type Diff<T extends string, U extends string> = ({ [P in T]: P } &
   { [P in U]: never } & { [x: string]: never })[T];
-export type Omit<T, K extends keyof T> = { [P in Diff<keyof T, K>]: T[P] };
+export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
```

Comment:
I think this is the more correct typing for `Omit`, and it is needed to make the above work. It looks like almost everywhere `Omit` is being used, it's being used as

```ts
Partial<Omit<..., ...>>
```

which seems like maybe it was a workaround for this?